From 1fd0f1bc6b2311c5c044bbdce6049f8dc2607b16 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:34 -0500
Subject: [drm] i915: Ignore VBT backlight presence check on Acer C720 (4005U)

Message-id: <1418852380-13061-24-git-send-email-rclark@redhat.com>
Patchwork-id: 102158
O-Subject: [RHEL7 drm 23/89] drm/i915: Ignore VBT backlight presence check on Acer C720 (4005U)
Bugzilla: 1153301
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1153301
Upstream: since v3.16.4

commit 3149c916e42b65aa57155aa294d9cec7ee8a7bf8

Author:     Scot Doyle <lkml14@scotdoyle.com>
AuthorDate: Thu Aug 21 16:08:02 2014 +0000
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:14:57 2014 -0500

    drm/i915: Ignore VBT backlight presence check on Acer C720 (4005U)

    commit dfb3d47b2369ad752ab9f7438bbf9949524b46ae upstream.

    commit c675949ec58ca50d5a3ae3c757892f1560f6e896
    Author: Jani Nikula <jani.nikula@intel.com>
    Date:   Wed Apr 9 11:31:37 2014 +0300

        drm/i915: do not setup backlight if not available according to VBT

    prevents backlight setup on the Acer C720 (Core i3 4005U CPU), which has a
    misconfigured VBT. Apply quirk to ignore the VBT backlight presence check
    during backlight setup.

    Signed-off-by: Scot Doyle <lkml14@scotdoyle.com>
    Tested-by: Tyler Cleveland <siralucardt@openmailbox.org>
    Cc: Jani Nikula <jani.nikula@intel.com>
    Cc: Daniel Vetter <daniel.vetter@ffwll.ch>
    Signed-off-by: Jani Nikula <jani.nikula@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index daf081d..6bacd64 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -11720,6 +11720,9 @@ static struct intel_quirk intel_quirks[] = {
  /* Acer C720 and C720P Chromebooks (Celeron 2955U) have backlights */
  { 0x0a06, 0x1025, 0x0a11, quirk_backlight_present },
 
+ /* Acer C720 Chromebook (Core i3 4005U) */
+ { 0x0a16, 0x1025, 0x0a11, quirk_backlight_present },
+
  /* Toshiba CB35 Chromebook (Celeron 2955U) */
  { 0x0a06, 0x1179, 0x0a88, quirk_backlight_present },
 
-- 
1.7.1